### PR TITLE
Feat: add get_by_id method

### PR DESCRIPTION
**Pull request by ischaojie**
Add `get_by_id` method for Model

### CODE REVIEW

File: mongox/models.py [hunk 1]
```diff
@@ -434,3 +434,15 @@ async def save(self: T) -> T:
         for k, v in self.dict(exclude={"id"}).items():
             setattr(self, k, v)
         return self
+
+    @classmethod
+    async def get_by_id(cls: typing.Type[T], id: str) -> T:
```
Comment (aminalaee) [inline mongox/models.py:439]: Can we also accept id as `bson.ObjectId`?
If yes, we can add and extend the tests for it.

File: mongox/models.py [hunk 2]
```diff
@@ -434,3 +434,15 @@ async def save(self: T) -> T:
...
+        """
+        Get document by id
+        """
+        id = bson.objectid.ObjectId(id)
+        try:
+            result = await cls.query({"_id": id}).get()
+        except NoMatchFound:
```
Comment (aminalaee) [inline mongox/models.py:446]: I guess this isn't needed, you raise whatever is raised anyway.

File: mongox/models.py [hunk 3]
```diff
@@ -434,3 +434,15 @@ async def save(self: T) -> T:
         for k, v in self.dict(exclude={"id"}).items():
             setattr(self, k, v)
         return self
+
+    @classmethod
+    async def get_by_id(cls: typing.Type[T], id: str) -> T:
+        """
+        Get document by id
+        """
+        id = bson.objectid.ObjectId(id)
```
Comment (aminalaee) [inline mongox/models.py:443]:
```suggestion
        """

        id = bson.objectid.ObjectId(id)
```

**Review by aminalaee — COMMENTED** [review]
Thank you for the PR. Just a few minor comments.

**Comment by ischaojie** [comment]
oh, I will fix it later

Sent with a Spark
2022年2月4日 +0800 17:52 Amin Alaee ***@***.***>，写道：

File: mongox/models.py [hunk 4]
```diff
@@ -434,3 +434,18 @@ async def save(self: T) -> T:
...
+    async def get_by_id(cls: typing.Type[T], id: typing.Union[str, bson.ObjectId]) -> T:
+        """
+        Get document by id
+        """
+
+        if isinstance(id, str):
+            try:
+                id = bson.ObjectId(id)
+            except bson.errors.InvalidId as e:
+                raise NoMatchFound() from e
```
Comment (ischaojie) [inline mongox/models.py:448]: I'm not sure should raise NoMatchFound when id invalid, for the user, the final result is same: item does not exist

Comment (aminalaee) [reply mongox/models.py:448]: Maybe worth creating another exception `InvalidObjectIdException(f'"{id}" is not a valid BSON ObjectId')`.
What do you think?

**Review by aminalaee — APPROVED** [review]
Thank you for the PR 👍

**Comment by angelhdzmultimedia** [comment]
Was this fixed? I'm getting an error that 'UserModel' has no attribute 'get_by_id'.
Thanks.

**Pull request merged**